packages/react-badge/src/components/Badge/useBadge.ts

```diff
@@ -1,29 +1,34 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
-import type { BadgeProps, BadgeState } from './Badge.types';
+import { getNativeElementProps, resolveShorthand, useMergedRefs } from '@fluentui/react-utilities';
+import type { BadgeProps, BadgeSlots, BadgeState } from './Badge.types';
 
 /**
  * Consts listing which props are shorthand props.
  */
-export const badgeShorthandPropsCompat = ['icon'] as const;
-
-const mergeProps = makeMergeProps<BadgeState>({ deepMerge: badgeShorthandPropsCompat });
+export const badgeSlots: Array<keyof BadgeSlots> = ['icon', 'root'];
 
 /**
  * Returns the props and state required to render the component
  */
-export const useBadge = (props: BadgeProps, ref: React.Ref<HTMLElement>, defaultProps?: BadgeProps): BadgeState => {
-  const state = mergeProps(
-    {
+export const useBadge = (props: BadgeProps, ref: React.Ref<HTMLElement>): BadgeState => {
+  const state: BadgeState = {
+    shape: 'circular',
+    size: 'medium',
+    iconPosition: 'before',
+    appearance: 'filled',
+    color: 'brand',
+    ...props,
```

Comment:
I'm fine with the alternatives, but it would mean that for each new prop you'd need to explicitly assign it to the state. the spread would just mean all 'custom' props are there. but you also get more props than you need that are spread so there's pros/cons to both 🤷

Reply:
I went with for destructuring `prop` I guess we'll just see which is better in the long term